doc/services/logging/index.rst

```diff
@@ -566,6 +565,130 @@ Logging v2
+
+    Domain IDs assigning example
+
+Lets consider a log message created on ``B_S`` domain. Initially, it will have
+``domain_id`` set to 0. When ``B_NS-B_S`` link receives the message it will
+increase the ``domain_id`` to 1 by adding ``B_NS-B_S`` offset. Then message is
+passed to ``A_NS``. When ``A_NS-B_NS`` links receives the message it will add
+offset (2) to the ``domain_id`` and message will end up with ``domain_id`` set
+to 3 which uniquely identifies message originator.
+
+Cross domain log message
+------------------------
+
+In majority of the cases address space of each domain is unique and one domain
+cannot access directly data in another domain. Because of that, backend may
+partially process the message before it is passed to another domain. Partial
+processing may include converting string package to ``fully self-contained``
+version (copying read-only strings to the package body).
+Each domain may have different timestamp source in terms of frequency and
+offset. Logging does not perform any timestamp conversion.
+
+Runtime filtering
+-----------------
+
+In single domain case each log source has dedicated variable with runtime
+filtering for each backend in the system. In multi-domain case, originator of
+the log message is not aware of number of backends in the root domain thus in
+order to filter logs in multiple domains, each source requires runtime
+filtering setting in each domain on the way to the root domain. Number of
+sources in other domain is not known during compilation thus runtime filtering
+of remote sources must use dynamically allocated memory (one word per each
+source). When a backend in the root domain changes filtering of the module from
+remote domain, local filter is updated. After the update, aggregate (maximum
+from all local backends) filter is checked and if changed then remote domain is
+informed about this change. With this approach, runtime filtering works the
+same way in multi-domain scenario as in single domain.
+
+Message ordering
+----------------
+
+Logging does not provide any mechanism for synchronizing timestamp across multiple
+domains thus if domains have different timestamp sources messages will be
+processed in the order of arrival to the buffer in the root domain.
+
+If domains have same timestamp source or there is an out of bound mechanism that
+recalculates timestamps then there are 2 options:
+
+- messages are processed as they arrive to the buffer in the root domain. Messages
+  are unordered but they can be sorted by the host as timestamp indicates time
+  of message generation.
+- links have dedicated buffers and during processing head of each buffer is checked
+  and oldest message is processed first. With this approach, it is possible to
+  maintain ordering of the messages at the cost of suboptimal RAM memory utilization
+  (since buffer is not shared) and increased processing latency (see
+  :kconfig:option:`CONFIG_LOG_PROCESSING_LATENCY_US`).
```

Comment:
```suggestion
Logging does not provide any mechanism for synchronizing timestamps across multiple
domains. If domains have different timestamp sources, messages will be
processed in the order of arrival to the buffer in the root domain.

If domains have the same timestamp source or if there is an out-of-bound mechanism that
recalculates timestamps, there are 2 options:

* Messages are processed as they arrive in the buffer in the root domain. Messages
  are unordered but they can be sorted by the host as the timestamp indicates the time
  of the message generation.
* Links have dedicated buffers. During processing, the head of each buffer is checked
  and the oldest message is processed first.
  With this approach, it is possible to maintain the order of the messages at the cost
  of a suboptimal memory utilization (since the buffer is not shared) and increased processing latency (see
  :kconfig:option:`CONFIG_LOG_PROCESSING_LATENCY_US`).
```